src/BuildIntegration/Microsoft.NETCore.Native.targets

```diff
@@ -282,11 +283,16 @@ See the LICENSE file in the project root for more information.
       <CustomLinkerArg Include="/OUT:&quot;$(NativeBinary)&quot;" Condition="'$(OS)' == 'Windows_NT'" />
       <CustomLinkerArg Include="/DEF:&quot;$(ExportsFile)&quot;" Condition="'$(OS)' == 'Windows_NT' and $(ExportsFile) != ''" />
       <CustomLinkerArg Include="/LIBPATH:&quot;%(AdditionalNativeLibraryDirectories.Identity)&quot;" Condition="'$(OS)' == 'Windows_NT' and '@(AdditionalNativeLibraryDirectories->Count())' &gt; 0" />
-      <CustomLinkerArg Include="-exported_symbols_list &quot;$(ExportsFile)&quot;" Condition="'$(OS)' != 'Windows_NT' and $(ExportsFile) != ''" />
+      <CustomLinkerArg Include="-exported_symbols_list &quot;$(ExportsFile)&quot;" Condition="'$(TargetOS)' == 'OSX' and $(ExportsFile) != ''" />
       <CustomLinkerArg Condition="Exists('$(_Win32ResFile)')" Include="&quot;$(_Win32ResFile)&quot;" />
       <CustomLinkerArg Include="@(LinkerArg)" />
     </ItemGroup>
-
+	<ItemGroup Condition="'$(OS)' != 'Windows_NT' and '$(TargetOS)' != 'OSX' and $(ExportsFile) != ''">
+		<CustomLinkerArg Include="-Wl,--version-script=$(ExportsFile)" />
+		<CustomLinkerArg Include="-Wl,--discard-all" />
+		<CustomLinkerArg Include="-Wl,--gc-sections" />
+		<CustomLinkerArg Include="-Wunused-command-line-argument" />
```

Comment:
Why do we need `-Wunused-command-line-argument` ? A comment may be nice.

Reply:
My mistake, I thought to long term use all these commands with --exported-symbols. The new version is incompatible with that.